ext/GradedArraysTensorAlgebraExt/GradedArraysTensorAlgebraExt.jl

```diff
@@ -1,109 +1,78 @@
+fuse_or_trivial_axis(::Tuple{}, axes::Tuple) = trivial_axis(axes)
 
-# Sort the blocks by sector and then merge the common sectors.
-function block_mergesort(a::AbstractArray)
-  I = blockmergesortperm.(axes(a))
-  return a[I...]
+function row_and_column_axes(
+  blocked_axes::BlockedTuple{2,<:Any,<:Tuple{Vararg{AbstractUnitRange}}}
+)
+  codomain_axes, domain_axes = blocks(blocked_axes)
+  @assert !(isempty(codomain_axes) && isempty(domain_axes))
+  row_axis = fuse_or_trivial_axis(codomain_axes, domain_axes)
+  unflipped_col_axis = fuse_or_trivial_axis(domain_axes, codomain_axes)
+  return row_axis, flip(unflipped_col_axis)
 end
 
-function TensorAlgebra.fusedims(
-  ::SectorFusion, a::AbstractArray, merged_axes::AbstractUnitRange...
+function TensorAlgebra.matricize(
+  ::SectorFusion, a::AbstractArray, biperm::AbstractBlockPermutation{2}
 )
-  # First perform a fusion using a block reshape.
-  # TODO avoid groupreducewhile. Require refactor of fusedims.
-  unmerged_axes = groupreducewhile(
-    unmerged_tensor_product, axes(a), length(merged_axes); init=OneToOne()
-  ) do i, axis
-    return length(axis) ≤ length(merged_axes[i])
-  end
-
-  a_reshaped = fusedims(BlockReshapeFusion(), a, unmerged_axes...)
+  a_perm = permutedims(a, Tuple(biperm))
+  row_axis, col_axis = row_and_column_axes(axes(a)[biperm])
+  a_reshaped = blockreshape(a_perm, (row_axis, col_axis))
   # Sort the blocks by sector and merge the equivalent sectors.
   return block_mergesort(a_reshaped)
 end
 
-function TensorAlgebra.splitdims(
-  ::SectorFusion, a::AbstractArray, split_axes::AbstractUnitRange...
+function TensorAlgebra.unmatricize(
+  ::SectorFusion,
+  m::AbstractMatrix,
+  blocked_axes::BlockedTuple{2,<:Any,<:Tuple{Vararg{AbstractUnitRange}}},
 )
   # First, fuse axes to get `blockmergesortperm`.
   # Then unpermute the blocks.
-  axes_prod = groupreducewhile(
-    unmerged_tensor_product, split_axes, ndims(a); init=OneToOne()
-  ) do i, axis
-    return length(axis) ≤ length(axes(a, i))
-  end
-  blockperms = blocksortperm.(axes_prod)
-  sorted_axes = map((r, I) -> only(axes(r[I])), axes_prod, blockperms)
+  row_col_axes = row_and_column_axes(blocked_axes)
+
+  blockperms = blocksortperm.(row_col_axes)
+  sorted_axes = map((r, I) -> only(axes(r[I])), row_col_axes, blockperms)
 
   # TODO: This is doing extra copies of the blocks,
   # use `@view a[axes_prod...]` instead.
   # That will require implementing some reindexing logic
   # for this combination of slicing.
-  a_unblocked = a[sorted_axes...]
-  a_blockpermed = a_unblocked[invblockperm.(blockperms)...]
-  return splitdims(BlockReshapeFusion(), a_blockpermed, split_axes...)
+  m_unblocked = m[sorted_axes...]
+  m_blockpermed = m_unblocked[invblockperm.(blockperms)...]
+  return unmatricize(FusionStyle(BlockSparseArray), m_blockpermed, blocked_axes)
 end
 
+# Sort the blocks by sector and then merge the common sectors.
+function block_mergesort(a::AbstractArray)
```

Comment:
```suggestion
function sectormergesort(a::AbstractArray)
```
In retrospect I think `blockmergesortperm` should have been called `sectormergesortperm` to make it clear it is merging and sorting by sector, but we can change that in another PR.